### PR TITLE
[BEAM-394] Updated the AvroCoder class to set unserializable fields as transient

### PR DESCRIPTION
**Pull request by ilganeli**
* Updated AvroCoder to mark non-serializable fields as transient.
* Updated the findbugs_filter to remove ignored errors.

### CODE REVIEW

**Comment by dhalperi** [comment]
Hi Ilya,

Thanks, this is great!

A few Checkstyle errors are breaking the build:

```
[ERROR] /Users/travis/build/apache/incubator-beam/sdks/java/core/src/main/java/org/apache/beam/sdk/coders/AvroCoder.java:171:21: 'final' modifier out of order with the JLS suggestions. [ModifierOrder]

[ERROR] /Users/travis/build/apache/incubator-beam/sdks/java/core/src/main/java/org/apache/beam/sdk/coders/AvroCoder.java:181:21: 'final' modifier out of order with the JLS suggestions. [ModifierOrder]

[ERROR] /Users/travis/build/apache/incubator-beam/sdks/java/core/src/main/java/org/apache/beam/sdk/coders/AvroCoder.java:182:21: 'final' modifier out of order with the JLS suggestions. [ModifierOrder]

[ERROR] /Users/travis/build/apache/incubator-beam/sdks/java/core/src/main/java/org/apache/beam/sdk/coders/AvroCoder.java:183:21: 'final' modifier out of order with the JLS suggestions. [ModifierOrder]

[ERROR] /Users/travis/build/apache/incubator-beam/sdks/java/core/src/main/java/org/apache/beam/sdk/coders/AvroCoder.java:184:21: 'final' modifier out of order with the JLS suggestions. [ModifierOrder]

[ERROR] /Users/travis/build/apache/incubator-beam/sdks/java/core/src/test/java/org/apache/beam/sdk/coders/AvroCoderTest.java:65:8: Unused import: java.io.IOException. [UnusedImports]

[ERROR] /Users/travis/build/apache/incubator-beam/sdks/java/core/src/test/java/org/apache/beam/sdk/coders/AvroCoderTest.java:68:8: Unused import: java.io.OutputStream. [UnusedImports]

[ERROR] /Users/travis/build/apache/incubator-beam/sdks/java/core/src/test/java/org/apache/beam/sdk/coders/AvroCoderTest.java:155: First sentence should end with a period. [JavadocStyle]

[ERROR] /Users/travis/build/apache/incubator-beam/sdks/java/core/src/test/java/org/apache/beam/sdk/coders/AvroCoderTest.java:156: Line is longer than 100 characters (found 111). [LineLength]

[ERROR] /Users/travis/build/apache/incubator-beam/sdks/java/core/src/test/java/org/apache/beam/sdk/coders/AvroCoderTest.java:162:3: '{' at column 3 should be on the previous line. [LeftCurly]

Audit done.

[INFO] There are 10 errors reported by Checkstyle 6.17 with beam/checkstyle.xml ruleset.
```

**Comment by ilganeli** [comment]
@dhalperi Thanks! Is there a checkstyle config somewhere I can import into IntelliJ?

**Comment by dhalperi** [comment]
Checkstyle config is in the build-tools somewhere -- should be easy to find. (Sorry, running off to a meeting). `checkstyle.xml`?

File: sdks/java/core/src/main/java/org/apache/beam/sdk/coders/AvroCoder.java [hunk 1]
```diff
@@ -178,10 +178,10 @@
   // Cache the old encoder/decoder and let the factories reuse them when possible. To be threadsafe,
   // these are ThreadLocal. This code does not need to be re-entrant as AvroCoder does not use
   // an inner coder.
-  private final ThreadLocal<BinaryDecoder> decoder;
-  private final ThreadLocal<BinaryEncoder> encoder;
-  private final ThreadLocal<DatumWriter<T>> writer;
-  private final ThreadLocal<DatumReader<T>> reader;
+  private final transient  ThreadLocal<BinaryDecoder> decoder;
```
Comment (dhalperi) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/coders/AvroCoder.java]: please remove extra space.

Comment (dhalperi) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/coders/AvroCoder.java]: Actually, I can do this in merge.

**Comment by dhalperi** [comment]
LGTM -- if you can fix the extra space great, or I can fix during the merge.

**Comment by dhalperi** [comment]
R: @dhalperi

Still LGTM, will merge when tests are green.

**Comment by dhalperi** [comment]
Seems like Travis was a flake, still merging.

**Pull request closed without merging**